lib/cocoapods/target/framework_paths.rb

```diff
@@ -0,0 +1,36 @@
+module Pod
+  class Target
+    class FrameworkPaths
+      # @return [String] the path to the .framework
+      #
+      attr_reader :source_path
+
+      # @return [String, Nil] the dSYM path, if one exists
+      #
+      attr_reader :dsym_path
+
+      def initialize(source_path, dsym_path = nil)
+        @source_path = source_path
+        @dsym_path = dsym_path
+      end
+
+      def ==(other)
+        if other.class == self.class
+          other.source_path == @source_path && other.dsym_path == @dsym_path
+        else
+          false
+        end
+      end
+
+      alias eql? ==
+
+      def hash
```

Comment:
do you use rubymine? I am curious if you were to generate this from the IDE what it would output?

Reply:
I do use rubymine but I'm not sure how to generate `hash` implementations

are you worried about the correctness of the implementation? for hashing the main requirement is if something is `==` to something else, their hashes must also be equal, but not the other way around. collisions are OK as long as `==` is correct

also since we're piggy backing on `Array`'s implementation we get it for free, if `Array`'s `hash` doesn't work then there are bigger problems :)